lib/ui/painting.dart

```diff
@@ -2886,6 +2886,15 @@ class Canvas extends NativeFieldWrapperClass2 {
   /// cause the new layer to be composited into the previous layer.
   void restore() native 'Canvas_restore';
 
+  /// Restore the current save stack to the state where [saveCount] is gotten.
+  ///
+  /// Use [save] and [saveLayer] to push state onto the stack, and use
+  /// [getSaveCount] to get the [saveCount].
+  ///
+  /// If a state was pushed with with [saveLayer], then this call will also
+  /// cause the new layer to be composited into the previous layer.
+  void restoreToCount(int saveCount) native 'Canvas_restoreToCount';
```

Comment:
I don't think we should have this API. It has implications that are very sketchy. For example, suppose you save, save, take the count, restore, restore, save, save, save, then restore to the count. You won't get the effect you want at all; that's clearly a bug, and it should fail, but it won't.

If we did keep this, we should define what happens when saveCount is greater than the current save count.